wordpress-mobile/WordPress-Android
# Android 12: Backup and restore - New format for the XML configuration

Fixes #16067

This PR adds backup configuration xml in the [new format](https://developer.android.com/about/versions/12/behavior-changes-12#new-format) for apps targetting Android 12. The [old configuration is still required for devices running Android 11 or lower](https://developer.android.com/about/versions/12/behavior-changes-12#manifest-flag), hence renamed and retained.

### Testing instructions
1. Update `targetSdkVersion` to 31 in `build.gradle`.
2. Switch to the flavor `vanillaDebug`.
3. Make sure backup is enabled on the device by running `adb shell bmgr enabled`. If not, enable it from `Settings` > `Google` -> `Backup`.
4. Run the app on an Android 12 device (make sure to log in and toggle through tabs so that at least `wordpress` `database` and `org.wordpress.android_preferences.xml` `sharedprefs` file is generated).
    To verify their existence, check the contents of the directory `/data/data/org.wordpress.android/shared_prefs` and confirm the existence of the file `org.wordpress.android_preferences.xml`, similarly check `/data/data/org.wordpress.android/databases` and confirm the existence of the file `wordpress` (you can use the tab "Device File Explorer" from Android Studio for this).
5. Run the command: `adb shell bmgr backupnow org.wordpress.android`. 
6. Once the backup succeeds, delete contents of of the directory `/data/data/org.wordpress.android/shared_prefs` and `/data/data/org.wordpress.android/databases`.
7. Make sure you don't run the app after this last step.
8. Run the command `adb shell dumpsys backup` and grab the token of the `Current` section.
9. Run the command `adb shell bmgr restore {token} org.wordpress.android` where the token is from the last step.
10. Check the contents of the directory `/data/data/org.wordpress.android/shared_prefs` and confirm that the file `org.wordpress.android_preferences.xml` is restored.
11. Check the contents of the directory `/data/data/org.wordpress.android/databases` and confirm that the file `wordpress` is restored.

For more details on the testing steps, check the official guide: https://developer.android.com/guide/topics/data/testingbackup

P.S. You may see a transport error on step 5. [This section](https://developer.android.com/guide/topics/data/testingbackup) can help in troubleshooting common issues. 
The first time I ran the backup command, I got this error:`Transport rejected package because it wasn't able to process it at the time`.
The issue was resolved on trying after [few hours](https://stackoverflow.com/questions/63364064/deciphering-transport-rejected-package-because-it-wasnt-able-to-process-it-at#comment114501726_63364064). 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
